pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -607,13 +607,57 @@ func validateMachinePools(cd *hivev1.ClusterDeployment) field.ErrorList {
 
 	vErrs = append(vErrs, validateMachinePool(cd, &cd.Spec.ControlPlane, field.NewPath("spec", "controlPlane"))...)
 
+	vErrs = append(vErrs, validateComputeMachinePoolRequirements(cd)...)
+
 	for i, mp := range cd.Spec.Compute {
 		vErrs = append(vErrs, validateMachinePool(cd, &mp, field.NewPath("spec", "compute").Index(i))...)
 	}
 
 	return vErrs
 }
 
+func validateComputeMachinePoolRequirements(cd *hivev1.ClusterDeployment) field.ErrorList {
```

Comment:
Pass in the slice of machine pools rather than the full clusterDeployment.

Reply:
i'll lose the platform info. while GCP happens to require MachinePool.Platform to be populated today (b/c we need the instance type) this may not always be the case

Reply:
Fair enough. Although I think it should **always** be the case that the Platform must be present matching the platform from the clusterDeployment. In fact, that should be validated by the webhook, too.